### PR TITLE
Settings for location accuracy!

### PR DESCRIPTION
**Pull request by JBassett**
Fixes: https://github.com/home-assistant/android/issues/868

### CODE REVIEW

File: app/src/full/java/io/homeassistant/companion/android/sensors/GeocodeSensorManager.kt [hunk 1]
```diff
@@ -58,7 +62,14 @@ class GeocodeSensorManager : SensorManager {
                     return@addOnSuccessListener
                 }
 
-                if (location.accuracy <= LocationSensorManager.MINIMUM_ACCURACY)
+                val sensorDao = AppDatabase.getInstance(context).sensorDao()
+                val sensorSettings = sensorDao.getSettings(geocodedLocation.id)
+                val minAccuracy = sensorSettings
+                    .firstOrNull { it.name == "Minimum Accuracy" }?.value?.toIntOrNull()
```
Comment (dshokouhi) [inline app/src/full/java/io/homeassistant/companion/android/sensors/GeocodeSensorManager.kt:68]: Should we replace the string with the constant here?

File: app/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -188,6 +189,7 @@ like to connect to:</string>
   <string name="sensor_description_internal_storage">Information about the total and available storage space internally</string>
   <string name="sensor_description_last_reboot">The date and time of the devices last reboot</string>
   <string name="sensor_description_light_sensor">The current level of illuminance</string>
+  <string name="sensor_description_location_accurate">Allow Home Assistant to send a notification to request an accurate location along with the application periodically requesting an accurate location.</string>
```
Comment (dshokouhi) [inline app/src/main/res/values/strings.xml:192]: Should we also describe the settings the user has control over?

File: app/src/full/java/io/homeassistant/companion/android/sensors/GeocodeSensorManager.kt (hunk 1 above)
Comment (JBassett) [reply app/src/full/java/io/homeassistant/companion/android/sensors/GeocodeSensorManager.kt]: Yep!

File: app/src/main/res/values/strings.xml (hunk 2 above)
Comment (JBassett) [reply app/src/main/res/values/strings.xml]: Done

**Review by dshokouhi — APPROVED** [review]
Looks good to me.

One question if a user has location permission granted the new sensor `singleAccurateLocation` is automatically enabled?

**Comment by JBassett** [comment]
> Looks good to me.
> 
> One question if a user has location permission granted the new sensor `singleAccurateLocation` is automatically enabled?

Yep!

**Comment by anyuta1166** [comment]
> Yep!

I don't think so. Initially sensors are disabled because we have no permissions. After that we explicitly enable zone and background sensors after getting permissions during initial setup.
So for **new** installations this sensor will be disabled by default.

**Comment by anyuta1166** [comment]
Why do you ever need a new sensor? Looks like another crutches. 

> Allow Home Assistant to send a notification to request an accurate location

This is used not only by Home Assistant notifications, but also triggered by any sensors update request, The description is misleading.

**Comment by JBassett** [comment]
> > Yep!
> 
> I don't think so. Initially sensors are disabled because we have no permissions. After that we explicitly enable zone and background sensors after getting permissions during initial setup.
> So for **new** installations this sensor will be disabled by default.

Feel free to test the change as I did while developing, it does indeed auto enable.


> Why do you ever need a new sensor? Looks like another crutches.

The thought behind the new sensor is for the minimum accuracy.  This allows users to specify what they specifically define as minimum accuracy for a background, zone, or accurate location independently from one another.


> This is used not only by Home Assistant notifications, but also triggered by any sensors update request, The description is misleading.

Feel free to open a PR to clarify the description, however this change only affect the location being requested/sent.  It has absolutely no affect on the other non location sensors.

**Pull request merged**